### PR TITLE
Zendesk go live notes

### PR DESCRIPTION
**Pull request by QuanNguyen-CDS**
# Summary | Résumé
[Trello](https://trello.com/c/AH8gsNqx/465-update-deal-in-zensales-with-go-live-request-notes)
feat: hooked in go live request into the already existing contact request rest API call
refactor: removed all the test and implementation code for freshdesk.create_ticket and the rest call `/<uuid:user_id>/support-email`
feat: updated zendesk sell with `create_note`, `search for deal`, and `create go live request`

# Test instructions | Instructions pour tester la modification
Validated that zendesk go live request populates zendesk sell
ran unit tests: `test_contact_request`, `test_zendesk_sell`, `test_freskdesk`, `test_rest`

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet
Note there a matching change on notification-admin which will come very shortly.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: app/clients/zendesk_sell.py
Comment (AntoineAugusti) [inline app/clients/zendesk_sell.py:121]:
```suggestion
    def _generate_note_data(resource_type: NoteResourceType, resource_id: str, content: str) -> Dict[str, str]:
```
I believe

File: app/user/rest.py
Comment (AntoineAugusti) [inline app/user/rest.py:465]: Can you add a test covering sending a go live request here?

File: app/clients/zendesk_sell.py
Comment (AntoineAugusti) [inline app/clients/zendesk_sell.py:309]: I believe you can just pass a `json` arg to `requests`, don't know if your private method accepts that

File: app/clients/freshdesk.py
Comment (AntoineAugusti) [inline app/clients/freshdesk.py:35]: I see that we use 2 `<br>` for demo requests, why is it different? What is the right thing?

Comment (QuanNguyen-CDS) [reply app/clients/freshdesk.py:35]: you are right, the question is why is the demo request different than the request to go live.
If you remember we moved the message generation from notification-admin into notification-api.
Functionally the message should be identical from the previous implementation going to freshdesk for the `demo` and `request to go live`. If we want to consider change the format of the message this is a change in functional logic at the freshdesk level

File: app/clients/zendesk_sell.py
Comment (QuanNguyen-CDS) [reply app/clients/zendesk_sell.py:121]: nice catch!

File: app/clients/freshdesk.py
Comment (AntoineAugusti) [reply app/clients/freshdesk.py:35]: Sounds good

**Review by AntoineAugusti — APPROVED** [review]
LGTM! Nice work.

I'll let you decide if you want to do more refactoring before merging otherwise it's all good

**Pull request merged**
